lib/Workflow/Base.pm

```diff
@@ -20,7 +20,11 @@ sub new {
     return $self;
 }
 
-sub init {return}
+sub init {return};
+
+sub log {
+    return ($_[0]->{log} ||=  Log::Log4perl->get_logger(ref $_[0]) );
```

Comment:
Cosmetic change required, the parentheses should use same spacing, so it does not look as something is missing.

```suggestion
return ( $_[0]->{log} ||=  Log::Log4perl->get_logger(ref $_[0]) );
````